src/core/function.js

```diff
@@ -364,7 +364,9 @@ var PDFFunction = (function PDFFunctionClosure() {
         var rmin = encode[2 * i];
         var rmax = encode[2 * i + 1];
 
-        tmpBuf[0] = rmin + (v - dmin) * (rmax - rmin) / (dmax - dmin);
+        // Ensure that this is always a valid number, since rounding errors
+        // may cause it to become NaN (fixes issue6113.pdf).
+        tmpBuf[0] = (rmin + (v - dmin) * (rmax - rmin) / (dmax - dmin)) || 0;
```

Comment:
I think we shall return value in range from rmin..rmax, can we use `rmin + (((v - dmin) * (rmax - rmin) / (dmax - dmin)) || 0);` here?

Also, it might be useful to check for 0 before div, e.g. `dmax === dmin` and return 0, e.g. `dmax === dmin ? rmin : (rmin + (v - dmin) * (rmax - rmin) / (dmax - dmin))`